tests/test-glib.c

```diff
@@ -824,7 +824,7 @@ static void test_logout_async(void) {
   snapd_client_set_auth_data(client, auth_data);
 
   g_assert_nonnull(mock_snapd_find_account_by_id(snapd, id));
-  AsyncData *data = async_data_new(loop, snapd);
+  g_autoptr(AsyncData) data = async_data_new(loop, snapd);
```

Comment:
This is a leak - the data is passed into the callback. (Other cases below).

Reply:
Yes, it is passed, but then we wait in a local mainloop(), so the data struct is freed after the g_mainloop_quit() called in the callback, when the control returns to the function that created the memory block. I think that this is more clear that having the creation in a function and the destruction in another one. So really there is no leak (in fact, there were two leaks before this PR because I created the memory block but forgot to free it in the callback; that's why I changed everything to g_autoptr).

Reply:
The main problem is that, sometimes, the callbacks are expected to be called several times, so it requires to think carefully when to free the passed data. Instead, using a g_autoptr() in the initial function, solves the problem automagically (as long as g_main_loop_run() is used, of course).

Reply:
Of course, the alternative is to remove the `g_autoptr()` that I added in the creation functions, and add an `async_data_free()` after each `g_main_loop_quit()` call in the callbacks, but in my opinion, that is more complex in this case.

Reply:
I understand now. I think we'd want to be careful doing it this way, as while it does make the code simpler, it's inconsistent with the other code.

Reply:
Mmm... I think that it is better to keep consistency... I'll revert this part.

Reply:
Ok, reverted and added some extra comments. It should be ready.